usaspending_api/common/api_request_utils.py

```diff
@@ -464,12 +466,16 @@ def build_response(self):
                     "matched_ids": Location.objects.filter(Q(**{"location_congressional_code": row[0], "location_state_code": row[1], "location_state_name": row[2]})).values_list("location_id", flat=True)
                 }
                 response_object.append(response_row)
+                if len(response_object) >= limit:
```

Comment:
To make sure we don't actually pull back more records than we need from the db, would it be possible to limit results when creating the queryset instead?

I'm guessing you might have wanted everything back so you could sort it. Any way to do sorting in the queryset as well? 

We're already seeing db performance degradation as we load more data, so I want to make sure we avoid transporting extra data/looping through results if we're able to perform the equivalent logic at the queryset level.